crates/observe/src/distributed_tracing/request_id.rs

```diff
@@ -28,6 +31,20 @@ use {
     tracing_subscriber::{Layer, Registry, layer::Context, registry::LookupSpan},
 };
 
+pub(crate) fn request_id(headers: &HeaderMap<HeaderValue>) -> String {
+    static INSTANCE: OnceCell<AtomicUsize> = OnceCell::new();
+    let counter = INSTANCE.get_or_init(|| AtomicUsize::new(0));
+
+    if let Some(header) = headers.get("X-Request-ID") {
+        String::from_utf8_lossy(header.as_bytes()).to_string()
+    } else {
+        format!(
+            "{}",
+            counter.fetch_add(1, std::sync::atomic::Ordering::SeqCst)
```

Comment:
`Relaxed` is actually sufficient but I didn't know that when writing the original code.
```suggestion
            counter.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
```